test/unit/BPool.t.sol

```diff
@@ -847,37 +847,192 @@ contract BPool_Unit_Bind is BasePoolTest {
 
-  function test_Set_TotalWeightIfDenormMoreThanOldWeight() private view {}
+    vm.prank(_caller);
+    vm.expectRevert('ERR_NOT_CONTROLLER');
+    bPool.rebind(_fuzz.token, _fuzz.balance, _fuzz.denorm);
+  }
 
-  function test_Set_TotalWeightIfDenormLessThanOldWeight() private view {}
+  function test_Revert_NotBound(address _token, Rebind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    vm.assume(_token != _fuzz.token);
 
-  function test_Revert_MaxTotalWeight() private view {}
+    vm.expectRevert('ERR_NOT_BOUND');
+    bPool.rebind(_token, _fuzz.balance, _fuzz.denorm);
+  }
 
-  function test_Set_Denorm() private view {}
+  function test_Revert_Finalized(Rebind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    _setFinalize(true);
 
-  function test_Set_Balance() private view {}
+    vm.expectRevert('ERR_IS_FINALIZED');
+    bPool.rebind(_fuzz.token, _fuzz.balance, _fuzz.denorm);
+  }
 
-  function test_Pull_IfBalanceMoreThanOldBalance() private view {}
+  function test_Revert_MinWeight(uint256 _denorm, Rebind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    vm.assume(_denorm < MIN_WEIGHT);
 
-  function test_Push_UnderlyingIfBalanceLessThanOldBalance() private view {}
+    vm.expectRevert('ERR_MIN_WEIGHT');
+    bPool.rebind(_fuzz.token, _fuzz.balance, _denorm);
+  }
 
-  function test_Push_FeeIfBalanceLessThanOldBalance() private view {}
+  function test_Revert_MaxWeight(uint256 _denorm, Rebind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    vm.assume(_denorm > MAX_WEIGHT);
 
-  function test_Emit_LogCall() private view {}
+    vm.expectRevert('ERR_MAX_WEIGHT');
+    bPool.rebind(_fuzz.token, _fuzz.balance, _denorm);
+  }
+
+  function test_Revert_MinBalance(uint256 _balance, Rebind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    vm.assume(_balance < MIN_BALANCE);
+
+    vm.expectRevert('ERR_MIN_BALANCE');
+    bPool.rebind(_fuzz.token, _balance, _fuzz.denorm);
+  }
+
+  function test_Revert_Reentrancy(Rebind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    // Assert that the contract is accessible
+    assertEq(bPool.call__mutex(), false);
+
+    // Simulate ongoing call to the contract
+    bPool.set__mutex(true);
+
+    vm.expectRevert('ERR_REENTRY');
+    bPool.rebind(_fuzz.token, _fuzz.balance, _fuzz.denorm);
+  }
+
+  function test_Set_TotalWeightIfDenormMoreThanOldWeight(Rebind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    vm.assume(_fuzz.denorm > _fuzz.previousDenorm);
+    bPool.rebind(_fuzz.token, _fuzz.balance, _fuzz.denorm);
+
+    assertEq(bPool.call__totalWeight(), _fuzz.totalWeight + (_fuzz.denorm - _fuzz.previousDenorm));
+  }
+
+  function test_Set_TotalWeightIfDenormLessThanOldWeight(Rebind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    vm.assume(_fuzz.denorm < _fuzz.previousDenorm);
+    bPool.rebind(_fuzz.token, _fuzz.balance, _fuzz.denorm);
+
+    assertEq(bPool.call__totalWeight(), _fuzz.totalWeight - (_fuzz.previousDenorm - _fuzz.denorm));
```

Comment:
The test cases above could be combined in the following way:

```solidity
function test_Set_TotalWeight(Rebind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
    bPool.rebind(_fuzz.token, _fuzz.balance, _fuzz.denorm);

    if (_fuzz.denorm > _fuzz.previousDenorm) {
        assertEq(bPool.call__totalWeight(), _fuzz.totalWeight + (_fuzz.denorm - _fuzz.
    } else if (_fuzz.denorm < _fuzz.previousDenorm) {
        assertEq(bPool.call__totalWeight(), _fuzz.totalWeight - (_fuzz.previousDenorm - _fuzz.denorm));
    }
}
```

However, yet I can't decide which alternative is preferable. What do you think?

Reply:
I think is clearer if we separate it in 2 different cases, because we are testing 2 different branches from an `if else`